subsys/bluetooth/host/iso.c

```diff
@@ -1383,7 +1383,14 @@ int bt_iso_big_terminate(struct bt_iso_big *big)
 void hci_le_big_complete(struct net_buf *buf)
 {
 	struct bt_hci_evt_le_big_complete *evt = (void *)buf->data;
-	struct bt_iso_big *big = &bigs[evt->big_handle];
+	struct bt_iso_big *big;
+
+	if (evt->big_handle > ARRAY_SIZE(bigs)) {
+		BT_WARN("Invalid BIG handle");
+		return;
+	}
```

Comment:
I suppose you are right, you know this better than me.

How do I do that?

Reply:
I guess we first need to find a way to mark a BIG as pending (there can only be a single BIG initiating or a single BIG syncing at one time (although we can have a BIG initiating and another BIG syncing at the same time), so I suggest adding a (atomic?) flag to the BIG struct with a `syncing` and a `initiating` bit and when this even arrives with an invalid handle, we iterate the array of BIGs and find the syncing/initiating (based on the event) and terminate it. 

To terminate an invalid BIG we would probably need to change the `struct bt_iso_big *big` parameter of `hci_le_terminate_big` and `hci_le_big_sync_term` to just be the (invalid) handle, and then also call `big_disconnect(big);` and `cleanup_big(big);` with the BIG we looked up so that we clean up the host as well as notifying the application.

Reply:
Is it correct now, I haven't done `big_disconnect` because other error handling in this function also did not do this. Is this something that needs to be improved?

Reply:
See https://github.com/zephyrproject-rtos/zephyr/pull/33461#discussion_r597510476